File: backend/services/core/src/test/java/app/mnema/core/config/RequiredSecretPropsTest.java
package app.mnema.core.config;

import app.mnema.core.media.config.MediaClientProps;
import app.mnema.core.security.CoreInternalAuthProps;
import jakarta.validation.Validation;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class RequiredSecretPropsTest {

    private final jakarta.validation.Validator validator = Validation.buildDefaultValidatorFactory().getValidator();

    @Test
    void rejectsMissingInternalServiceTokens() {
        assertThat(validator.validate(new MediaClientProps("http://media.test", " ")))
                .extracting(violation -> violation.getPropertyPath().toString())
                .containsExactly("internalToken");
        assertThat(validator.validate(new CoreInternalAuthProps("")))
                .extracting(violation -> violation.getPropertyPath().toString())
                .containsExactly("internalToken");
    }
}

File: backend/services/user/src/test/kotlin/app/mnema/user/config/RequiredSecretPropsTest.kt
package app.mnema.user.config

import app.mnema.user.media.config.MediaClientProps
import app.mnema.user.security.UserInternalAuthProps
import jakarta.validation.Validation
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class RequiredSecretPropsTest {
    private val validator = Validation.buildDefaultValidatorFactory().validator

    @Test
    fun `rejects missing media and user service tokens`() {
        val mediaViolations = validator.validate(MediaClientProps("http://media.test", ""))
        val userViolations = validator.validate(UserInternalAuthProps(" "))

        assertEquals(setOf("internalToken"), mediaViolations.map { it.propertyPath.toString() }.toSet())
        assertEquals(setOf("internalToken"), userViolations.map { it.propertyPath.toString() }.toSet())
    }
}
